madmom/audio/spectrogram.py

```diff
@@ -1235,3 +1238,106 @@ def process(self, data, **kwargs):
             data, crossover_frequencies=self.crossover_frequencies,
             fmin=self.fmin, fmax=self.fmax, norm_filters=self.norm_filters,
             unique_filters=self.unique_filters, **kwargs)
+
+
+class SemitoneBandpassSpectrogram(np.ndarray):
+    """
+    Construct a semitone spectrogram by using time domain filters as
+    described in [1]_.
+
+    Parameters
+    ----------
+    filename : str
+        Name of the audio file or file handle.
```

Comment:
I do not like `filename`, please use `signal` since you basically expect a `Signal` here (which takes care of loading the audio file).